<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Generate simulated data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singlecell-de</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Generate simulated data</h1>

</div>


<p><strong>Last updated:</strong> 2017-10-19</p>
<p><strong>Code version:</strong> 861184a54d0981caa0791fd3a31ddee6995fa00b</p>
<hr />
<div id="current-considerations" class="section level2">
<h2>Current considerations</h2>
<ol style="list-style-type: decimal">
<li>SCDE requires at least 2000 genes to run and require filtering…</li>
</ol>
<hr />
</div>
<div id="simulation-strategy" class="section level2">
<h2>Simulation strategy</h2>
<p>We will compare and evaluate methods based simulated datasets that are generated based on real datasets of single cell RNA-seq and bulk RNA-seq. These simulated datasets will permit us to evaluate the power performance of the methods while controlling for important parameters: number of sequencing samples, number of genes, depth of coverage, fraction of true DE genes. Most importantly, our simulation approach preserves the sample correlation structure in each sequencing datasets and thereby provides “realistic” evaluation datasets.</p>
<p>We will generate three major classes of simulated datasets:<br />
1. single cell RNA-sequencing datasets, including both UMI and read-based count data<br />
2. bulk RNA-sequencing data, using GTEx liver samples<br />
3. thinned bulk RNA-sequencing data</p>
<p>The simulation parameters are:<br />
1. Number of sequencing samples/group: n = 40, 80, 200, 500 (the upper limit is fixed for each dataset; expect higher number for drop-seq and at most 90~ for Fluidigm C1)<br />
2. Number of genes: G = 1,000<br />
3. Number of simulated datasets: M = 50<br />
4. Fraction of null genes: <span class="math inline">\(\pi_0 \in \{ .5, .9, 1\}\)</span><br />
5. Thinning parameter: <span class="math inline">\(\theta \in {.001, .01}\)</span> 6. Effect size distribution: <span class="math inline">\(\hat{\beta}_g \sim N(0, 1)\)</span></p>
<p>Here we briefly summarize the steps in simulated data generation: 1. Take a subset of <span class="math inline">\(G\)</span> genes,<br />
2. Take a subset of <span class="math inline">\(2n\)</span> samples and assign the samples to two arbitrary biological conditions: either same labels at each gene (preserve correlation structure) or different labels at each gene (no sample correlation),<br />
3. Modify the expression count data to add signal.</p>
</div>
<div id="add-signals" class="section level2">
<h2>Add signals</h2>
<p>The method of adding signals has been previously described: ash (?), vash (?), and Gerard and Stephens (RUV and mouthwash). For the motivation of this approach, see the main text in Gerard and Stephens (RUV) for a detailed description. The overview here closely follows the text in Gerard and Stephens (RUV).</p>
<p>We add signals to the null datasets in a randomly selected fraction of genes <span class="math inline">\(\pi_1 = 1 - \pi0\)</span> (<span class="math inline">\(\pi_1 = .1\)</span> or <span class="math inline">\(.5\)</span>). The signals <span class="math inline">\(\beta_g\)</span> are randomly sampled from a <span class="math inline">\(N(0, ?)\)</span>. the standard deviation was chosen so that the classification would be neither too easy nor too hard (?). Let</p>
<p><span class="math inline">\(\beta_{g1}, \dots, \beta_{g \pi_1*p} \sim N(0, s^2)\)</span></p>
<p>be the effect sizes, where <span class="math inline">\(gl \in \Omega\)</span>, the set of non-null genes. Then we drew a <span class="math inline">\(\mathbf{Z}\)</span> matrix of the same dimension as the expression count matrix <span class="math inline">\(\mathbf{Y}\)</span> by</p>
<p><span class="math display">\[
z_{igl} | y_{igl} \sim 
\begin{cases}
\text{Binomial}(y_{igl}, 2^{\beta_{gl} x_{i}}) &amp; \text{if $\beta_{gl} &lt; 0$ and $gl \in \Omega$},\\
\text{Binomial}(y_{igl}, 2^{-\beta_{gl} (1-x_{i})}) &amp; \text{if $\beta_{gl} &gt; 0$ and $gl \in \Omega$},\\
\delta(y_{igl}) &amp; \text{if $jl \notin \Omega$}, 
\end{cases}
\]</span></p>
<p>where <span class="math inline">\(\delta(y_{igl})\)</span> indicates a point mass at <span class="math inline">\(y_{igl}\)</span>. We then use <span class="math inline">\(\mathbf{Z}\)</span> as our non-null expression count matrix.</p>
<div id="how-to-choose-s2" class="section level3">
<h3>How to choose <span class="math inline">\(s^2\)</span>?</h3>
<p>As a rule of thumb, large <span class="math inline">\(s^2\)</span> generates large effect sizes and thus DE classification is easier. To help choosing the value of <span class="math inline">\(s^2\)</span>, I compute the fraction of true positive genes (fraction of true signals classifed as DE) given 5% and 1% false positive rate (proportion of nulls classifed as DE) under the parameter setting:</p>
<ol style="list-style-type: decimal">
<li>Number of sequencing samples per group: 50 and 100<br />
</li>
<li>Number of genes: 1000<br />
</li>
<li>Number of simulated data: 50</li>
<li>Fraction of null genes: .5, .9</li>
</ol>
<p><code>limmaVoom</code> was used to perform DE analysis. Overall, a stringent FDR threshold makes classification more difficult (lower TPR compared when 1% FDR compared to 5% FDR) under different sample size, fraction of null genes, and <span class="math inline">\(s^2\)</span>. About the parameter settings, all but fraction of null genes may affect TPR under fixed FDR. Larger sample size is associated with higher TPR. More importantly, a large <span class="math inline">\(s^2\)</span> is associated with high TPR - which is as expected since there are more large effect sizes.</p>
<p>When choosing <span class="math inline">\(s^2\)</span>, the rule of thumb is setting a value of false positve rate that makes the classification neither easy nor difficult. We chose a value of <span class="math inline">\(s^2\)</span> such taht TPR is about 50%. (maybe report results under other <span class="math inline">\(s^2\)</span> values in the supplemental materials). For now, we choose 1% FDR and <span class="math inline">\(s = 1\)</span>.</p>
<pre class="r"><code>make_tpr_plot(tpr01_long, title = &quot;TPR given 1% FDR&quot;)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 3.4.2</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<p><img src="figure/benchmark-simulation.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>make_tpr_plot(tpr05_long, title = &quot;TPR given 5% FDR&quot;)</code></pre>
<p><img src="figure/benchmark-simulation.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="how-to-choose-parameter-settings-for-different-distributional-assumptions" class="section level3">
<h3>How to choose parameter settings for different distributional assumptions?</h3>
<p>question: what parameter values are required for shape A and shape B to have the same TPR under fixed FDR, sample size, and pi0</p>
<hr />
</div>
</div>
<div id="other-papers" class="section level2">
<h2>Other papers</h2>
<p>In Soneson and Robinson (2017), the simulated datasets include a subset of single cell samples but all of the genes in the original datasets.</p>
<hr />
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre><code>R version 3.4.1 (2017-06-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2  dplyr_0.7.4   ggplot2_2.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.13     bindr_0.1        knitr_1.17       magrittr_1.5    
 [5] munsell_0.4.3    colorspace_1.3-2 R6_2.2.2         rlang_0.1.2     
 [9] stringr_1.2.0    plyr_1.8.4       tools_3.4.1      grid_3.4.1      
[13] gtable_0.2.0     htmltools_0.3.6  yaml_2.1.14      lazyeval_0.2.0  
[17] rprojroot_1.2    digest_0.6.12    assertthat_0.2.0 tibble_1.3.4    
[21] glue_1.1.1       evaluate_0.10.1  rmarkdown_1.6    labeling_0.3    
[25] stringi_1.1.5    compiler_3.4.1   scales_0.5.0     backports_1.1.1 
[29] pkgconfig_2.0.1 </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
